src/server/frontend_wayland/window_wl_surface_role.h

```diff
@@ -102,8 +109,16 @@ class WindowWlSurfaceRole : public WlSurfaceRole
     OutputManager* output_manager;
     std::shared_ptr<WlSurfaceEventSink> const sink;
     std::unique_ptr<scene::SurfaceCreationParameters> const params;
-    std::experimental::optional<geometry::Size> pending_window_size;
-    std::experimental::optional<geometry::Size> committed_window_size;
+
+    /// The explicitly set (not taken from the surface buffer size) uncommitted window size
+    std::experimental::optional<geometry::Size> pending_explicit_size;
+
+    /// If the committed window size was set explicitly, rather than being taken from the buffer size
+    bool committed_size_set_explicitly{false};
```

Comment:
Having both optional `committed_size` _and_ a boolean to describe how to interpret it is a bit jarring. It smells of a missing abstraction but I don't have a better idea right now.